QLogger.cpp

```diff
@@ -1,266 +1,264 @@
+      if (!mModuleDest.contains(module))
       {
-         const auto log = new QLoggerWriter(fileDest, level);
-         log->stop(mIsStop);
+         const auto log = initializeWriter(fileDest, level, fileFolderDestination,
+                                           mode, fileSuffixIfFull, messageOptions);
+
+         mModuleDest.insert(module, log);
+
+         if (!mIsStop)
+         {
+            if (notify) {
+                const auto threadId
+                        = QString("%1").arg((quintptr)QThread::currentThread(), QT_POINTER_SIZE * 2, 16, QChar('0'));
+                log->enqueue(QDateTime::currentDateTime(), threadId, module, LogLevel::Info, "", "", -1, "Adding destination!");
+            }
+            if (mode != LogMode::Disabled)
+              log->start();
+         }
 
-         moduleDest.insert(module, log);
          allAdded = true;
       }
    }
 
    return allAdded;
 }
 
-void QLoggerManager::queueMessage(const QString module, const QVector<QVariant> &logData)
+QLoggerWriter *QLoggerManager::initializeWriter(const QString &fileDest, LogLevel level,
+                                                const QString &fileFolderDestination, LogMode mode,
+                                                LogFileDisplay fileSuffixIfFull, LogMessageDisplays messageOptions) const
 {
-   if (mNonWriterQueue.count(module) < QUEUE_LIMIT)
-      mNonWriterQueue.insert(module, logData);
+    const QString lFileDest                  = fileDest.isEmpty() ? mDefaultFileDestination : fileDest;
+    const LogLevel lLevel                    = level == LogLevel::Warning ? mDefaultLevel : level;
+    const QString lFileFolderDestination     = fileFolderDestination.isEmpty() ? mDefaultFileDestinationFolder : QDir::fromNativeSeparators(fileFolderDestination);
+    const LogMode lMode                      = mode == LogMode::OnlyFile ? mDefaultMode : mode;
+    const LogFileDisplay lFileSuffixIfFull   = fileSuffixIfFull == LogFileDisplay::DateTime ? mDefaultFileSuffixIfFull : fileSuffixIfFull;
+    const LogMessageDisplays lMessageOptions = messageOptions.testFlag(LogMessageDisplay::Default) ? mDefaultMessageOptions : messageOptions;
+
+    const auto log = new QLoggerWriter(lFileDest, lLevel, lFileFolderDestination,
+                                       lMode, lFileSuffixIfFull, lMessageOptions);
+
+    log->setMaxFileSize(mDefaultMaxFileSize);
+    log->stop(mIsStop);
+
+    return log;
+}
+
+void QLoggerManager::clearFileDestinationFolder(const QString &fileFolderDestination, int days)
+{
+    QDir dir(fileFolderDestination + QStringLiteral("/logs"));
+    if (!dir.exists()) {
+        return;
+    }
+
+    dir.setFilter(QDir::Files | QDir::Hidden | QDir::NoSymLinks);
+    const QFileInfoList list = dir.entryInfoList();
+
+    const QDateTime now = QDateTime::currentDateTime();
+
+    QFileInfoList::const_iterator i;
+    for (i = list.constBegin(); i != list.constEnd(); ++i) {
+        const QFileInfo fileInfo = *i;
```

Comment:
const auto&

Although probably here is not going to be a problem with Qt allocating a new object since we're using a const iterator, is a just in case.

The other option would be to use the pointer with the name:

`for (auto fileInfoIter = list.constBegin(); fileInfoIter != list.constEnd(); ++fileInfoIter) {`

and then `fileInfoIter->lastModified()....`

Reply:
Yes you right!